taskmanager/src/Application/Report/Report.php

```diff
@@ -0,0 +1,62 @@
+<?php declare(strict_types=1);
+
+namespace App\Application\Report;
+
+use App\Application\Task\Task;
+use App\Application\Task\TaskRegistry;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+
+class Report
+{
+    private $id;
+    private $name;
+    private $criteria;
+    private $elements;
+    private $tasks;
+
+    public function __construct(string $name, Criteria $criteria)
+    {
+        $this->id = Uuid::uuid4();
+        $this->name = $name;
+        $this->criteria = $criteria;
+        $this->elements = [];
+        $this->tasks = [];
+    }
+
+    public function getId(): UuidInterface
+    {
+        return $this->id;
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function generate(TaskRegistry $tasks)
+    {
+        //@TODO: foreach criteria get tasks and add to elements or tasks
+        // how to decide if assigne tasks or elements like Efficiency | 33,3% hmm.. ?
```

Comment:
@tarnawski @partikus @mateuszmaslanka @dgrochowski Hmm.. that's the most important question for that PR. And no idea how to solve 🤔 Comments more than appreciated.

Reply:
Maybe we should simplify the logic of generating reports and create one type of report contains task filtered by criteria.